src/network/local_socket_reply.cpp

```diff
@@ -139,38 +143,105 @@ void mp::LocalSocketReply::send_request(const QNetworkRequest& request, QIODevic
     auto op = request.attribute(QNetworkRequest::CustomVerbAttribute).toByteArray();
 
     // Build the HTTP method part
-    http_data += op;
-    http_data += ' ';
-    http_data += request.url().toString();
+    http_data += op + ' ' + request.url().path();
+
+    if (request.url().hasQuery())
+    {
+        http_data += "?" + request.url().query();
+    }
+
     http_data += " HTTP/1.1\r\n";
 
     // Build the HTTP Host header
-    // Host can be anything, so we'll use 'multipass'
-    http_data += "Host: multipass\r\n";
+    auto host = request.url().host().toUtf8();
+    if (!host.isEmpty())
+    {
+        http_data += "Host: " + host + "\r\n";
+    }
+    else
+    {
+        throw mp::HttpLocalSocketException("Host is required in URL");
+    }
 
     // Build the HTTP User-Agent header
-    http_data += "User-Agent: Multipass/";
-    http_data += mp::version_string;
-    http_data += "\r\n";
+    auto user_agent = request.header(QNetworkRequest::UserAgentHeader).toByteArray();
+    if (!user_agent.isEmpty())
+    {
+        http_data += "User-Agent: " + user_agent + "\r\n";
+    }
+
+    if (!local_socket_write(http_data))
+        return;
 
     if (op == "POST" || op == "PUT")
     {
-        http_data += "Content-Type: application/x-www-form-urlencoded\r\n";
+        http_data = "Content-Type: " + request.header(QNetworkRequest::ContentTypeHeader).toByteArray() + "\r\n";
 
-        if (outgoingData)
+        if (outgoingData && outgoingData->size() > 0)
         {
+            auto content_length = request.header(QNetworkRequest::ContentLengthHeader).toByteArray();
+            auto transfer_encoding = request.rawHeader("Transfer-Encoding");
+            bool is_chunked{false};
+
+            if (!content_length.isEmpty() && !transfer_encoding.isEmpty())
+            {
+                throw mp::HttpLocalSocketException(
+                    "Both \'Content-Length\' and \'Transfer-Encoding\' headers cannot be set at the same time");
+            }
```

Comment:
I don't think that's true… You can't have it `chunked` and provide `Content-Length`, but you could have [any other value of `Transfer-Encoding`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Transfer-Encoding).

Unless you mean that it's not supported, in which case reword the exception? :)

Reply:
Ah, ok, I misread the specs.  Then this check needs to be for `chunked` instead of `Transfer-Encoding` and I'll reword the error message.

Reply:
Done.